Insecticide & friends

nexusmrsep opened the pull request:
#### Summary
```SUMMARY: Content "Insecticide and ways to deploy it"```

#### Purpose of change
Having in mind that there are many commercial insecticides/pesticides and we have abundance of bug roaming around recently, I though that some chemical extermination would be nice to implement.

Somewhat inspired on Vormithrax's exterminator playthrough.

#### Describe the solution
In this PR I've taken fungicide as a template and:
- added insecticide (with recipe), that doesn't do anything by itself, but is a base for further gaseous deployment recipes
- added insecticidal gas as ammo for chemical thrower
- added insecticide gas grenade (no recipe - reserved for future chemical plants/factories and/or faction production)
- added makeshift insecticide gas grenade (with recipe)
- added insecticidal gas field - the final effect of above, deals damage to insect species

Also revisited fungicide and:
- added fungicidal gas grenade (no recipe - reserved for future chemical plants/factories and/or faction production)
- added fungicidal gas grenade (with recipe)
More ways to effectively fight ever spreading fungus is always welcomed.

#### Describe alternatives you've considered
- Should insecticide work on spider species too?
- Should the insecticide have less fictional ingredients? What would they be?

#### Additional context
N/A, however during the tests I noticed that existing gas spreading mechanics might be a bit underwhelming for smoke grenades. Might need quick re-balancing.

NickGrease commented:
Is there a 'rule' for when json lists (eg. components list) should be on a single line vs multiple lines?

anothersimulacrum commented:
Lines in the JSON files should not be longer than 120 characters. The linter enforces that, so if there are lines longer than they, the file should be linted.

Reclusive-reptile commented:
Nicotine and its derivatives are a common real world insecticide, if you are looking for suggestions.

Night-Pryanik commented:
> Should insecticide work on spider species too?

Despite spiders ain't insects, quick googling revealed that broad-spectrum insecticides should work on them too.

matskuman5 commented on data/json/items/chemicals_and_resources.json:
```diff
@@ -284,6 +284,31 @@
+    "comestible_type": "FOOD",
+    "symbol": "!",
+    "healthy": -7,
+    "description": "Agricultural grade chemical insecticide powder designed to eradicate insect pests.",
```
```suggestion
    "description": "Agricultural-grade chemical insecticide powder designed to eradicate insect pests.",
```

Amoebka commented:
A bit pedantic, but insecticides and fungicides shouldn't be "gases". "Aerosol" describes them better, because the actual poison is liquid/solid, with the gas acting merely as propellant.

roaringjohn commented on data/json/items/grenades.json:
```diff
@@ -287,5 +287,93 @@
+    "countdown_action": { "type": "transform", "target": "canister_empty" },
+    "flags": [ "TRADER_AVOID" ]
+  },
+    {
```
```suggestion
  {
```

Amoebka commented:
And another thing. Do insects/spiders currently try to path around/run away from clouds of insecticide? They should, since it's a contact variety ("burns your skin" and immediate damage).

nexusmrsep commented:
@Amoebka 
I've changed the game descriptions to "haze", "mist", or "spray" where applicable. However the in-code descriptions will be still referring to it as a being a gas because it uses gas-spread mechanic.

As for insects avoiding this field: if an insect is intelligent (MF_AVOID_DANGER_2 flag for AI) it should avoid the field. I haven't checked if any insect has this flag.

@Night-Pryanik 
I've added SPIDER species to the list of creatures affected.

@Reclusive-reptile 
I've added another recipe based on extracting nicotine from tobacco, tobacco products, and nicotine liquid for ecigs.

Amoebka commented:
Insects and spiders currently don't have any PATH_AVOID_DANGER at all, which is wrong in itself, since animal intelligence is PATH_AVOID_DANGER_1 (all mammals have it). I would also argue that animal intelligence should be enough to avoid deadly gas clouds, because, well, real life animals do that.

anothersimulacrum commented on data/json/items/ammo/chemical_spray.json:
```diff
@@ -34,5 +34,23 @@
+    "name": "sprayable insecticide",
+    "description": "Strong sprayable insecticide ammo for the chemical thrower.  Best be used with some kind of mask or mouth protection.",
+    "weight": 3,
+    "volume": 1,
```
```suggestion
    "volume": "250 ml",
```

ZhilkinSerg commented on src/map_field.cpp:
```diff
@@ -1264,6 +1267,10 @@ bool map::process_fields_in_submap( submap *const current_submap,
                         furn_set( p, f_null );
                     }
                 }
+                if( curtype == fd_insecticidal_gas ) {
```
This is should be defined in json (and it is actually defined, but with different values):

```JSON
    "percent_spread": 100,
    "outdoor_age_speedup": "1 minutes",
    "dirty_transparency_cache": true,
```
